src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java

```diff
@@ -142,7 +143,7 @@ public void rip() throws IOException {
     }
     private String fileNameFromURL(URL url) {
         String saveAs = url.toExternalForm();
-        if (saveAs.substring(saveAs.length() - 1) == "/") { saveAs = saveAs.substring(0,saveAs.length() - 1) ;}
+        if (Objects.equals(saveAs.substring(saveAs.length() - 1), "/")) { saveAs = saveAs.substring(0,saveAs.length() - 1) ;}
```

Comment:
shouldn't `String#equals` (i.e. `a.equals(b)`) be applicable here?

Reply:
(Same comment applies to all other usages of `Objects.equals`)

Reply:
> shouldn't String#equals (i.e. a.equals(b)) be applicable here?

I've no idea. I don't know all that much about Java

Reply:
They are both `String`s so I think we should do `<leftstring>.equals(<rightstring>)` in all these cases. (Which calls the `String#equals` method.)